Themes/default/languages/Install.english.php

```diff
@@ -231,10 +262,70 @@
 $txt['upgrade_time_elapsed'] = 'Time Elapsed';
 $txt['upgrade_time_mins'] = 'mins';
 $txt['upgrade_time_secs'] = 'seconds';
+$txt['upgrade_username'] = 'Username:';
+$txt['upgrade_wrong_username'] = 'Username Incorrect';
+$txt['upgrade_password'] = 'Password:';
+$txt['upgrade_wrong_password'] = 'Password Incorrect';
 
+$txt['upgrade_wait'] = 'Please wait while a backup is created. For large forums this may take some time!';
+$txt['upgrade_wait2'] = 'Please wait while your database is converted to UTF-8. For large forums this may take some time!';
+$txt['upgrade_sec_login'] = 'For security purposes please login with your admin account to proceed with the upgrade.';
 $txt['upgrade_incomplete'] = 'Incomplete';
 $txt['upgrade_not_quite_done'] = 'Not quite done yet!';
 $txt['upgrade_paused_overload'] = 'This upgrade has been paused to avoid overloading your server. Don\'t worry, nothing\'s wrong - simply click the <label for="contbutt">continue button</label> below to keep going.';
+$txt['upgrade_continue_step'] = 'Continue from step reached during last execution of upgrade script.';
+$txt['upgrade_bypass'] = '<strong>Note:</strong> If necessary the above security check can be bypassed for users who may administrate a server but not have admin rights on the forum. In order to bypass the above check simply open &quot;upgrade.php&quot; in a text editor and replace &quot;$disable_security = false;&quot; with &quot;$disable_security = true;&quot; and refresh this page.';
+$txt['upgrade_areyouready'] = 'Before the upgrade gets underway please review the options below - and hit continue when you\'re ready to begin.';
+$txt['upgrade_backup_table'] = 'Backup tables in your database with the prefix';
+$txt['upgrade_recommended'] = 'recommended!';
+$txt['upgrade_maintenace'] = 'Put the forum into maintenance mode during upgrade.';
+$txt['upgrade_maintenance_title'] = 'Maintenace Title:';
+$txt['upgrade_maintenace_message'] = 'Maintenance Message:';
+$txt['upgrade_customize'] = 'Customize';
+$txt['upgrade_debug_info'] = 'Output extra debugging information';
+$txt['upgrade_empty_errlog'] = 'Empty error log before upgrading';
+$txt['upgrade_delete_karma'] = 'Delete all karma settings and info from the DB';
+$txt['upgrade_stats_collection'] = 'Allow Simple Machines to Collect Basic Stats Monthly.';
+$txt['upgrade_stats_info'] = 'If enabled, this will allow Simple Machines to visit your site once a month to collect basic statistics. This will help us make decisions as to which configurations to optimise the software for. For more information please visit our <a href="https://www.simplemachines.org/about/stats.php" target="_blank" rel="noopener">info page';
```

Comment:
Ew. This one is all kinds of nasty. If the opening tag is included in a $txt string, the closing tag should be, too.
Also, this should probably use sprintf to insert the URL rather than hard coding it.